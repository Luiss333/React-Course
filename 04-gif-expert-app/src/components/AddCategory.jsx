import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault(); //evita que se recargue la p√°gina al enviar el formulario (comportamiento por defecto)
    // console.log(inputValue);
    if (inputValue.trim().length <= 1) return; //trim() elimina los espacios en blanco al inicio y al final
    //setCategories((categories) => [inputValue, ...categories]);
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <>
      <form onSubmit={(event) => onSubmit(event)}>
        <input
          type="text"
          placeholder="Buscar gifs"
          value={inputValue}
          onChange={onInputChange}
        />
      </form>
    </>
  );
};
