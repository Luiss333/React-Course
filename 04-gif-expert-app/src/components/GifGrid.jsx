//import { useState, useEffect } from "react";
// import { getGifs } from "../helpers/getGifs";
import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  //console.log({ isLoading });
  //--------------------------------------------
  // const [images, setImages] = useState([]);

  // const getImages = async () => {
  //   const newImages = await getGifs(category);
  //   setImages(newImages);
  // };

  // useEffect(() => {
  //   getImages();
  // }, []);

  return (
    <>
      <h3>{category}</h3>
      <div className="card-grid">
        {images.map((image) => (
          <GifItem
            key={image.id}
            {...image} // Spread operator para enviar todas las propiedades de la imagen como argumentos
          />
        ))}
      </div>
    </>
  );
};
