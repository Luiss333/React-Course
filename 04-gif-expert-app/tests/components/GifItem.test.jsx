import { render } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas en <GifItem.jsx/>", () => {
  const title = "Luffy";
  const url = "https://one-piece.com/luffy.jpg";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });
});
