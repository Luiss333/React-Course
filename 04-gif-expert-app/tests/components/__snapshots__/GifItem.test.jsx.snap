// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem.jsx/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Luffy"
      src="https://one-piece.com/luffy.jpg"
    />
    <p>
      Luffy
    </p>
  </div>
</div>
`;
